lib/modules/datasource/docker/index.ts

```diff
@@ -548,6 +631,80 @@ export class DockerDatasource extends Datasource {
     return null;
   }
 
+  @cache({
+    namespace: 'datasource-docker-architecture',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      currentDigest: string
+    ) => `${registryHost}:${dockerRepository}@${currentDigest}`,
+    ttlMinutes: 1440,
+  })
+  public async getImageArchitecture(
+    registryHost: string,
+    dockerRepository: string,
+    currentDigest: string
+  ): Promise<string | null> {
```

Comment:
```suggestion
  ): Promise<string | null | undefined> {
```

We should return undefined on error, so we don't cache them. Errors are highly unexpected here.
Maybe we shouldn't do any error handling here at all? 🤔 
The only error we can expect is a 404, so the manifest is no longer available.